### ISSUE TITLE
node-red data persistence fails with permissions error

### ISSUE THREAD

**Issue by greg-mcnamara** [k8s-at-home/charts#1470]
# Details

**Helm chart name and version:**
node-red:10.0.0

**Container name and tag:**
nodered/node-red:2.1.4

**What steps did you take and what happened:**
I'm trying to deploy node-red to AWS EKS with Fargate and persistent volume claim to an EFS access point. When the pod starts up it crashes with the following error in the logs:

```
> node-red-docker@2.1.4 start /usr/src/node-red
> node $NODE_OPTIONS node_modules/node-red/red.js $FLOWS "--userDir" "/data"

internal/fs/utils.js:332
    throw err;
    ^

Error: EACCES: permission denied, copyfile '/usr/src/node-red/node_modules/node-red/settings.js' -> '/data/settings.js'
    at Object.copyFileSync (fs.js:2061:3)
    at copyFile (/usr/src/node-red/node_modules/fs-extra/lib/copy-sync/copy-sync.js:70:6)
    at onFile (/usr/src/node-red/node_modules/fs-extra/lib/copy-sync/copy-sync.js:56:25)
    at getStats (/usr/src/node-red/node_modules/fs-extra/lib/copy-sync/copy-sync.js:48:44)
    at handleFilterAndCopy (/usr/src/node-red/node_modules/fs-extra/lib/copy-sync/copy-sync.js:33:10)
    at Object.copySync (/usr/src/node-red/node_modules/fs-extra/lib/copy-sync/copy-sync.js:26:10)
    at Object.<anonymous> (/usr/src/node-red/node_modules/node-red/red.js:125:20)
    at Module._compile (internal/modules/cjs/loader.js:1085:14)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1114:10)
    at Module.load (internal/modules/cjs/loader.js:950:32) {
  errno: -13,
  syscall: 'copyfile',
  code: 'EACCES',
  path: '/usr/src/node-red/node_modules/node-red/settings.js',
  dest: '/data/settings.js'
}
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! node-red-docker@2.1.4 start: `node $NODE_OPTIONS node_modules/node-red/red.js $FLOWS "--userDir" "/data"`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the node-red-docker@2.1.4 start script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.
```

**Relevant Helm values:**

I'm installing via command line as below. I added the security context settings after reading #537 which appears to be the same issue.
```
helm install node-red \
  --set env.NODE_RED_ENABLE_PROJECTS="true" \
  --set persistence.data.enabled="true" \
  --set persistence.data.existingClaim="pvc-nodered" \
  --set podSecurityContext.runAsUser=1000 \
  --set podSecurityContext.runAsGroup=1000 \
  --set podSecurityContext.fsGroup=1000 \
  --set securityContext.runAsUser=1000 \
  --set securityContext.runAsGroup=1000 \
    k8s-at-home/node-red
```

**What did you expect to happen:**
The install would use the persistent volume claim for data persistence and would have sufficient rights on the volume to create the necessary files.

**Anything else you would like to add:**
I have also configured my AWS EFS access point as shown below:
![Capture_EFS_perms_node-red](https://user-images.githubusercontent.com/94930614/160475362-4dc13d6c-01d8-429c-b7d0-07d85d696f5c.PNG)

**Additional Information:**

This issue has been reported here and on node-red docker forums. Some have apparently solved the issue by running chown 1000:1000 on the volume attached to the docker container, however this is not practical or feasible on kubernetes.

**Comment by onedr0p** [comment]
Very intriguing, this is the first time I've seen someone try to deploy this in the cloud using EKS. Have you tried using EBS instead of EFS?

**Comment by greg-mcnamara** [comment]
Hi @onedr0p, I'm using EKS Fargate (serverless pods, no EC2) which doesn't support EBS yet. I opened a ticket on the node-red community forum, but after seeing #537 I thought someone here might be able to help. The same node-red issue has been reported multiple times across various forums, mostly relating to docker deployments. I haven't had any problems with other containers and EFS, so it must be something to do with the way they've configured the user in the image which is not kubernetes-friendly.

**Comment by onedr0p** [comment]
You've got guts running stateful workloads in a stateless environment like fargate. Not all workloads will be happy using a shared file store (EFS/NFS) for their persistent data too. You can add ec2 nodes (bottlerocket, SSH-less, immutable) to your existing EKS cluster that are able to consume EBS while having fargate as well. You can use both to their best ability.

**Comment by greg-mcnamara** [comment]
Yes I must admit it's all a bit experimental! Anyway, I managed to get it working with help from the node-red community [here](https://discourse.nodered.org/t/running-node-red-on-kubernetes-file-permissions-error-on-pod-initialization/60420/10). It was either my PV/PVC config, or changing the EFS root directory from default "/" to a named folder. Either way if anyone else has the same issue hopefully they won't lose a day like I did... Re: Fargate, aside from not paying for excessive headroom on EC2 and EBS (I always seemed to over-cater my node groups and volumes), I like the multi-zone availability you get with Fargate + EFS, after a slightly painful setup (e.g. creating mount points for each EFS in each subnet). I'm sure I'll hit something that must use EC2 (e.g. anything with a daemonset) but so far so good. Thanks for your help!
